accessibility_monitoring_platform/apps/cases/forms.py

```diff
@@ -182,6 +189,26 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.fields["sector"].empty_label = "Unknown"  # type: ignore
 
+    def clean_previous_case_url(self):
+        """Check url contains case number"""
+        previous_case_url = self.cleaned_data.get("previous_case_url")
+        if not previous_case_url:
+            return previous_case_url
+        if requests.head(previous_case_url).status_code >= 400:
+            raise ValidationError("Previous case URL does not exist")
+        try:
+            case_id: str = re.search(".*/cases/(.+?)/view/?", previous_case_url).group(  # type: ignore
+                1
+            )
+        except AttributeError:
+            raise ValidationError(  # pylint: disable=raise-missing-from
+                "Previous case URL did not contain case id"
+            )
+        if case_id.isdigit() and Case.objects.filter(id=case_id).exists():
+            return previous_case_url
+        else:
+            raise ValidationError("Previous case not found in platform")
```

Comment:
Might be worth writing some inline comments explaining what each step is checking for

Reply:
I figured that the error message text explained what was going on.

Reply:
Done. f1254dd891004fc42b81d43fb572151f8aeadbf3